torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -810,19 +789,26 @@ def _post_backward_hook(
                     else unsharded_grad
                 )
                 new_sharded_grad = torch.empty_like(chunks[0])  # padded
-                state._comm_hook(
-                    state._comm_hook_state,
-                    padded_unsharded_grad,
-                    new_sharded_grad,
-                )
-                if handle._sharding_strategy in (
-                    HandleShardingStrategy.HYBRID_SHARD,
-                    HandleShardingStrategy._HYBRID_SHARD_ZERO2,
-                ):
-                    default_hooks.allreduce_hook(
-                        state=state._inter_node_state,
-                        grad=new_sharded_grad,
+
+                if state._comm_hook is None:  # default path
+                    _div_if_needed(padded_unsharded_grad, prediv_factor)
+                    dist.reduce_scatter_tensor(
+                        new_sharded_grad,
+                        padded_unsharded_grad,
+                        group=state.process_group,
+                    )
+                    if not uses_hybrid_sharded_strategy:
+                        _div_if_needed(new_sharded_grad, postdiv_factor)
+                    if uses_hybrid_sharded_strategy:
+                        dist.all_reduce(new_sharded_grad, group=state._inter_node_pg)
+                        # TODO: Refactor this if we make the all-reduce async
+                        _div_if_needed(new_sharded_grad, postdiv_factor)
```

Comment:
might result in less duplication:

```
if uses_hybrid:
    all_reduce()
_div_if_needed(new sharded grad, postdiv factor)
```

because both situations call _div_if_needed?

Reply:
Yep, thanks!